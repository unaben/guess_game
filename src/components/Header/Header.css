h1, .guess {
  margin-bottom: 1rem;
}


