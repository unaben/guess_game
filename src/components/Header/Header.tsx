import { message } from "../../translate/ENT";
import "./Header.css";

const Header = () => {
  return (
    <>
      <h1>{message.titleText}</h1>
      <p className="guess" data-testid="guess">{message.guessText}</p>
    </>
  );
};
export default Header;
