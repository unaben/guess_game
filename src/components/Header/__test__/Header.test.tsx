import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../Header";

describe("Header Component", () => {
  it("renders correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it("renders heading text", () => {
    render(<Header />);
    const headingEl = screen.getByRole("heading", { level: 1 });
    expect(headingEl).toBeInTheDocument();
    expect(headingEl).toHaveTextContent(/Let's play a numbers game/i);
  });

  it("renders paragraph text", () => {
    render(<Header />);
    const paragraphEl = screen.getByTestId("guess");
    expect(paragraphEl).toBeInTheDocument();
    expect(paragraphEl).toHaveTextContent(/Guess a number between 1 and 30/i);
  });
});
