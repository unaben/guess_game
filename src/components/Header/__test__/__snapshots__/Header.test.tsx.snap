// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<div>
  <h1>
    Let's play a numbers game
  </h1>
  <p
    class="guess"
    data-testid="guess"
  >
    Guess a number between 1 and 30
  </p>
</div>
`;
